bin/create-release-tag

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+
+if [[ $# -ne 1 ]]; then
+    echo "usage: ${0##*/} (edge|stable)-xx.xx.xx" >&2
+    exit 1
+fi
+
+new_linkerd_version="$1"
+
+# Verify the tag format
+tag_format="^(test|edge|stable)-([0-9]+)\.([0-9]+)\.([0-9]+)$"
+if [[ $new_linkerd_version =~ $tag_format ]]; then
+    # todo: Use these values to verify the tag version increment.
+    # release_channel="${BASH_REMATCH[1]}"
+    # release_major="${BASH_REMATCH[2]}"
+    # release_minor="${BASH_REMATCH[3]}"
+    # release_patch="${BASH_REMATCH[4]}"
+    :
+else
+    echo "tag format incorrect; expected: $tag_format"
+    echo "example: edge-20.12.2, stable-2.10.1"
+    exit 1
+fi
+
+# todo: Verify the tag version increment.
+
+rootdir=$( cd "${0%/*}"/.. && pwd )
+
+# Make temporary file to save the release commit message into.
+tmp=$(mktemp -t release-commit-message.XXX.txt)
+echo $tmp
+
+# Save commit message into temporary file.
+#
+# Match each occurence of the regex and increment `n` by 1. While n == 1
+# (which is true only for the first section) print that line of `CHANGES.md`.
+# This ends up being the first section of release changes.
+awk '/^## (edge|stable)-[0-9]+\.[0-9]+\.[0-9]+/{n++} n==1' $rootdir/CHANGES.md > $tmp
+
+# Create an unsigned tag with the commit message.
```

Comment:
As said by the comment, this is creating an unsigned tag object right now. `-s` requires the user to have GPG setup, and nothing about the release workflow currently relies on tags being signed.

Reply:
So signing a tag object with `-s` requires the user have GPG signing setup. I don't actually have this; I use the "signed-off-by" `-s` for commits but nothing for tags.

As part of this script merging, we could now require any member responsible for an edge or stable release needs setup GPG signing, but since we don't actually do anything with verifying the tag I thought that not be necessary to add right now.

So pushing unsigned tags is not changing anything about the way that tags are signed or not right now--they could be either. If we start using `-s` here then you need GPG signing setup.

Reply:
Okay after discussion this morning, `-s` is now added to the `git tag` command. Members responsible for releases will be responsible for being able to GPG sign the newly created tags.